amethyst_network/src/bundle.rs

```diff
@@ -0,0 +1,105 @@
+use super::{ConnectionManagerSystem, NetConnectionPool, NetIdentity, NetReceiveBuffer,
+            NetSendBuffer, NetSocketSystem};
+use amethyst_core::bundle::{ECSBundle, Result};
+use filter::NetFilter;
+use serde::Serialize;
+use serde::de::DeserializeOwned;
+use shred::DispatcherBuilder;
+use specs::World;
+use std::net::SocketAddr;
+use uuid::Uuid;
+
+/// A convenience bundle to create the infrastructure needed to send and receive network messages.
+pub struct NetworkBundle<'a, T> {
+    /// The local ip to bind to.
+    ip: &'a str,
+    /// The local port to bind to.
+    port: Option<u16>,
+    /// The filters applied on received network events.
+    filters: Vec<Box<NetFilter<T>>>,
+    /// Indicates if this should behaves as a server or as a client when handling remote connections.
+    is_server: bool,
+    /// The server to automatically connect to.
+    /// You would usually want this if you set is_server = false.
+    connect_to: Option<SocketAddr>,
+}
+
+impl<'a, T> NetworkBundle<'a, T> {
+    /// Creates a new NetworkBundle in client mode.
+    pub fn new_client(
+        ip: &'a str,
+        port: Option<u16>,
+        filters: Vec<Box<NetFilter<T>>>,
+    ) -> Self {
+        NetworkBundle {
+            ip,
+            port,
+            filters,
+            is_server: false,
+            connect_to: None,
+        }
+    }
+
+    /// Creates a new NetworkBundle in server mode
+    pub fn new_server(
+        ip: &'a str,
+        port: Option<u16>,
+        filters: Vec<Box<NetFilter<T>>>,
+    ) -> Self {
+        NetworkBundle {
+            ip,
+            port,
+            filters,
+            is_server: true,
+            connect_to: None,
+        }
+    }
+
+    /// Automatically connects to the specified client network socket address.
+    pub fn with_connect(mut self, socket: SocketAddr) -> Self {
```

Comment:
Any reason why this isn't on the client constructor?

Reply:
Maybe you want to create the network socket when launching the game, but you don't want to connect to the server before the player presses the "connect to server" button in their game.

Reply:
Check.